src/base/bittorrent/peerinfo.cpp

```diff
@@ -198,7 +198,16 @@ QString PeerInfo::I2PAddress() const
 
 QString PeerInfo::client() const
 {
-    return QString::fromStdString(m_nativeInfo.client);
+    QString client = QString::fromStdString(m_nativeInfo.client);
+
+    for(QChar& c : client)
+    {
+        // replace non-printable characters with spaces
+        if (!c.isPrint())
+            c = u' ';
+    }
+
+    return client;
```

Comment:
Shouldn't we also remove leading/trailing spaces?
```suggestion
    return client.trimmed();
```

Reply:
Hmm, maybe. It could improve readability in some cases.

Reply:
if you do it then better do it right after `QString::fromStdString()` to shorten the iteration over the string.

Reply:
Iteration could introduce additional whitespaces, which we also want to trim.

Reply:
Actually `QString::simplified()` may be even better. It collapses internal whitespaces too.